src/Validation/Validator.php

```diff
@@ -556,9 +573,13 @@ protected function _processRules($field, ValidationSet $rules, $data, $newRecord
                 continue;
             }
 
-            $errors[$name] = $message;
             if (is_string($result)) {
                 $errors[$name] = $result;
+            } elseif ($this->_useI18n) {
+                $args = $rule->get('pass');
+                $errors[$name] = __d($this->_validationDomain, $name, $this->_translateArgs($args));
```

Comment:
I don't get why this is necessary. Validators are now defined in methods which can easily use the i18n features to translate messages. Having the validation messages magically translated has a few drawbacks:

* Hidden behavior that is not explicit.
* No way to opt out of translations.
* Validation messages cannot be extracted with the other i18n tools.